### ISSUE TITLE
pyright is missing from optional dependencies + wrong `check` env dependencies

### ISSUE THREAD

**Issue by benoit74** [openzim/_python-bootstrap#6]
In pyproject.toml, pyright is not present as an optional dependencies.

And hatch `check` environment depends from `test` optional dependencies which it does not use.

**Comment by rgaudin** [comment]
We could also have the wrapper in a _feature_ and advertise it should pyright call failed due to missing binary.

**Comment by benoit74** [comment]
I understand most of your concerns, but (yes, there is a but 🤣) 
- not using the wrapper also means in many cases not having a consistent pyright version between CI and dev environment ; this is made especially complex when jumping from one project to another one
- the wrapper install node and pyright only if missing
- the wrapper install pyright in a well known location, either `PYRIGHT_PYTHON_CACHE_DIR` if set, or else`XDG_CACHE_HOME` if set, or else `~/.cache` ; this avoids having many times the same version which is meaninfull
- the wrapper does not automatically updates pyright, it keeps in sync the wrapper and the tool versions and warns you that a new version is available (which might be a bit annoying, but ok)
```
> pyright --version
WARNING: there is a new pyright version available (v1.1.312 -> v1.1.317).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

pyright 1.1.312
```
- the wrapper works well when offline, timeout is set at 1 sec (obviously two stuff are tried since it took 2 secs in my test)
```
> time pyright
0 errors, 0 warnings, 0 informations
pyright  2.02s user 0.15s system 146% cpu 1.474 total
```

This first point is a significant concern for me, as a developer I usually prefer to not mind about such details, and as a maintainer I would prefer to not have to explain this to every new developer on a project. 

And yes, I added the wrapper in a _feature_ in my PR.

**Comment by rgaudin** [comment]
OK I'm sold 😉 Thanks for the details. I clearly misread the README.
